src/main/groovy/wooga/gradle/version/VersionPluginExtension.groovy

```diff
@@ -241,14 +242,19 @@ trait VersionPluginExtension implements BaseSpec {
         scope = value
     }
 
+
     /**
-     * @return Used during nearest version inference
+     * @return Prefix to be used when storing versions. Useful to distinguish projects in a same repository
      */
-    TagStrategy getTagStrategy() {
-        tagStrategy
+    Property<String> prefix = objects.property(String)
```

Comment:
You should use the same template @Azurelol and I use for setting up gradle properties.
In case of an extension you can just use a groovy public field. But in case of tasks gradle wants the @Input to be put on a getter and that the field is marked private. Since we use traits in a lot of places to share these properties and implementations etc we tend to use the same setup everywhere.

```groovy
private final Property<$TYPE$> $NAME$ = objects.property($TYPE$)

$INPUT_TYPE$
Property<$TYPE$> get$U_NAME$() {
    $NAME$
}

void set$U_NAME$(Provider<$TYPE$> value) {
    $NAME$.set(value)
}

void set$U_NAME$($TYPE$ value) {
    $NAME$.set(value)
}
```

Reply:
Also addressed in #60. I have my own template for this as well, I guess we should put stuff together and make some shared templates for the team?